### PR TITLE
Induction motor control

### PR DESCRIPTION
**Pull request by fbook98**

### CODE REVIEW

File: gem_controllers/gem_controller.py [hunk 1]
```diff
@@ -47,7 +47,8 @@ def make(
              speed controller.
             base_current_controller('PI'/'PID'/'P'/'ThreePoint'): Selection of the basic control algorithm for the
              current controller.
-            a(float):
+            a(float): Design parameter of the controller.
```
Comment (atra94) [inline gem_controllers/gem_controller.py]: That description is meaningless. Is there a correct technical term for this parameter? (@max-schenke ?)

File: gem_controllers/stages/emf_feedforward_ind.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from .emf_feedforward import EMFFeedforward
```
Comment (atra94) [inline gem_controllers/stages/emf_feedforward_ind.py:1]: Code documentation missing

File: gem_controllers/stages/operation_point_selection/scim_ops.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
+import gym_electric_motor as gem
```
Comment (atra94) [inline gem_controllers/stages/operation_point_selection/scim_ops.py:1]: Code documentation missing

File: gem_controllers/stages/operation_point_selection/scim_ops.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+import numpy as np
+import scipy.interpolate as sp_interpolate
+import matplotlib.pyplot as plt
+from .operation_point_selection import OperationPointSelection
+from ..base_controllers import PIController
+from ..anti_windup import AntiWindup
+
+
+class SCIMOperationPointSelection(OperationPointSelection):
```
Comment (atra94) [inline gem_controllers/stages/operation_point_selection/scim_ops.py]: This class looks quite similar to the PMSM Operation Point selection class.

Wouldn't it be beneficial to use a common base class (e.g. FieldOrientedControllerOperationPointSelection)?

**Review by atra94 — CHANGES_REQUESTED** [review]
A lot of good work in this PR.

I know that in the whole repository a lot of documentation is missing. Nonetheless, this PR should be documented with the same standards as in the gym-electric-motor repo.

File: gem_controllers/gem_controller.py (hunk 1 above)
Comment (max-schenke) [reply gem_controllers/gem_controller.py]: I assume that 'a' refers to the tuning parameter of the symmetrical optimum design method. If that is the case, I would be satisfied with naming it that way. Additionally, one could refer to some literature source where this method is explained,

File: gem_controllers/stages/emf_feedforward_ind.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    def tune(self, env, env_id, **_):
+        super().tune(env, env_id, **_)
+        mp = env.physical_system.electrical_motor.motor_parameter
+        self.r_r = mp['r_r']
+        self.l_m = mp['l_m']
+        self.l_r = mp['l_sigr'] + self.l_m
+        self.i_sq_idx = env.state_names.index('i_sq')
+        self.psi_abs_idx = env.state_names.index('psi_abs')
+
+    def reset(self):
```
Comment (atra94) [inline gem_controllers/stages/emf_feedforward_ind.py]: The reset method here is redundant

**Review by max-schenke — COMMENTED** [review]
I have no immediate concerns about the code but some more in-line comments would be helpful for the future.
Thank you very much and keep up the good work!
Please address @atra94 's concerns about missing documentation.

File: example.py [hunk 6]
```diff
@@ -1,23 +1,25 @@
...
+
+env_id = 'AbcCont-TC-SCIM-v0'
 
-env_id = 'DqCont-SC-PMSM-v0'
 env = gem.make(
-    env_id,
-    visualization=gem.visualization.MotorDashboard(state_plots='all')
-)
+        env_id,
+        state_action_processors=(FluxObserver(),),
```
Comment (atra94) [inline example.py:13]: Add a comment that the Flux Observer is required for the controller

File: gem_controllers/stages/operation_point_selection/foc_operation_point_selection.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+
+class FieldOrientedControllerOperationPointSelection(OperationPointSelection):
+    """
+    This is the base class for all field-oriented operating point controls. It also includes a function for level
+    control.
+    """
+
+    def __init__(self, max_modulation_level: float = 2 / np.sqrt(3), modulation_damping: float = 1.2):
+        self.mp = None
```
Comment (atra94) [inline gem_controllers/stages/operation_point_selection/foc_operation_point_selection.py:20]: doc missing

File: gem_controllers/stages/operation_point_selection/foc_operation_point_selection.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+        self.k_ = None
+        self.limited = None
+        self.u_dc = None
+        self.integrated = 0
+        self.psi_high = None
+        self.psi_low = None
+        self.integrated_reset = None
+
+    def tune(self, env, env_id, current_safety_margin=0.2):
+        super().tune(env, env_id, current_safety_margin)
```
Comment (atra94) [inline gem_controllers/stages/operation_point_selection/foc_operation_point_selection.py:56]: doc missing

**Review by atra94 — CHANGES_REQUESTED** [review]
Still too few documentation in some files

File: example.py (hunk 6 above)
Comment (atra94) [reply example.py:13]: Still not resolved

**Review by atra94 — APPROVED** [review]

**Pull request merged**